src/Native/Runtime/amd64/ManagedCalloutThunk.S

```diff
@@ -0,0 +1,56 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+.intel_syntax noprefix
+#include <unixasmmacros.inc>
+
+//
+// Defines a small assembly thunk designed to be used when unmanaged code in the runtime calls out to managed
+// code. In such cases the stack walker needs to be able to bridge the unmanaged gap in the stack between the
+// callout and whatever managed code initially entered the runtime. This thunk makes that goal achievable by
+// (a) exporting a well-known address in the thunk that will be the result of unwinding from the callout (so
+// the stack frame iterator knows when its hit this case) and (b) placing a copy of a pointer to a transition
+// frame saved when the previous managed caller entered the runtime into a well-known location relative to the
+// thunk's frame, enabling the stack frame iterator to recover the transition frame address and use it to
+// re-initialize the stack walk at the previous managed caller.
+//
+// If we end up with more cases of this (currently it's used only for the ICastable extension point for
+// interface dispatch) then we might decide to produce a general routine which can handle an arbitrary number
+// of arguments to the target method. For now we'll just implement the case we need, which takes two regular
+// arguments (that's the 2 in the ManagedCallout2 name).
+//
+// Inputs:
+//      rdi : Argument 1 to target method
+//      rsi : Argument 2 to target method
+//      rdx : Target method address
+//      rcx : Pointer to previous managed method's transition frame into the runtime
+//
+NESTED_ENTRY ManagedCallout2, _TEXT, NoHandler
+
+        // Push an rbp frame. Apart from making it easier to walk the stack the stack frame iterator locates
+        // the transition frame for the previous managed caller relative to the frame pointer to keep the code
+        // architecture independent.
+        push_nonvol_reg rbp
+        set_cfa_register rbp, 0
```

Comment:
`set_cfa_register` is not required here because of the frame size is fixed. We do not have it for all other fixed size frames, this one should not be special.